server-spi-private/src/main/java/org/keycloak/models/utils/KeycloakModelUtils.java

```diff
@@ -1188,4 +1188,31 @@ public static void setDefaultGroups(KeycloakSession session, RealmModel realm, S
     public static boolean isAdminPermissionsEnabled(RealmModel realm) {
         return Profile.isFeatureEnabled(Profile.Feature.ADMIN_FINE_GRAINED_AUTHZ_V2) && realm.isAdminPermissionsEnabled();
     }
+
+    public static void setupAdminPermissionsClient(KeycloakSession session, RealmModel realm, String clientId) {
+        ClientModel client = session.clients().addClient(realm, clientId);
+        realm.setAdminPermissionsClient(client);
+        RepresentationToModel.createResourceServer(client, session, false);
+    }
+
+    public static void setupAdminPermissionsClient(KeycloakSession session, RealmModel realm) {
+        setupAdminPermissionsClient(session, realm, determineAdminPermissionsClientId(session, realm));
+    }
+
+    private static String determineAdminPermissionsClientId(KeycloakSession session, RealmModel realm) {
```

Comment:
I was thinking more about this and to avoid potential conflicts with existing clients we could just append the UUID to the client id or something similar?

Reply:
For now we will use hardcoded version of clientId. Potential collision names should be solved within follow-up issue.